src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedTagStore.java

```diff
@@ -115,6 +135,35 @@ public boolean containsKey(final long index, final String key)
         return false;
     }
 
+    @Override
+    public boolean equals(final Object other)
+    {
+        if (other instanceof PackedTagStore)
+        {
+            if (this == other)
+            {
+                return true;
+            }
+            final PackedTagStore that = (PackedTagStore) other;
+            if (!this.keys.equals(that.keys))
+            {
+                return false;
+            }
+            if (!this.values.equals(that.values))
+            {
+                return false;
+            }
+            // We use reference equality for the dictionary, since PackedTagStores do not actually
```

Comment:
Could you elaborate more on when do you think the references should be equal? Maybe an example in the comment might make it better to understand

Reply:
Good idea. A lot of the equals/hashCode implementations are still in flux.